components/brave_ads/core/internal/serving/new_tab_page_ad_serving.cc

```diff
@@ -107,7 +106,6 @@ void NewTabPageAdServing::ServeAd(
     return FailedToServeAd(std::move(callback));
   }
 
-  CHECK(eligible_ads_);
```

Comment:
`CHECK` is not required because we call `IsSupported` within the flow of `MaybeServeAd`